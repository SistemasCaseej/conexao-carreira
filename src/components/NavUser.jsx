"use client"

import {ChevronsUpDown, LogOut, User} from "lucide-react"
import { Avatar} from "@/components/ui/avatar"
import { DropdownMenu, DropdownMenuContent, DropdownMenuItem, DropdownMenuLabel, DropdownMenuSeparator, DropdownMenuTrigger} from "@/components/ui/dropdown-menu"
import { SidebarMenu, SidebarMenuButton, SidebarMenuItem, useSidebar} from "@/components/ui/sidebar"
import { useRouter } from "next/navigation";

export function NavUser({user}) {
    const { isMobile } = useSidebar()
    const router = useRouter();

    const handleLogout = async () =>{

        try {
            const response = await fetch("/api/session", {
                method: "DELETE",
            });

            if (response.ok) {
                router.push("/candidate-login");
            } else {
                console.error("Erro ao deslogar:", await response.text());
            }
        }catch (error){
            console.error("Erro ao deslogar:", error);
        }
    }

    return (
        <SidebarMenu>
            <SidebarMenuItem>
                <DropdownMenu>
                    <DropdownMenuTrigger asChild className="cursor-pointer hover:bg-white hover:text-[#4c1286]">
                        <SidebarMenuButton size="lg">
                            <Avatar className="h-8 w-8 rounded-lg">
                                <User/>
                            </Avatar>
                            <div className="grid flex-1 text-left text-sm leading-tight">
                                <span className="truncate font-medium">{user?.name || "Usuário"}</span>
                                <span className="truncate text-xs">{user?.email || "user@gmail.com"}</span>
                            </div>
                            <ChevronsUpDown className="ml-auto size-4" />
                        </SidebarMenuButton>
                    </DropdownMenuTrigger>
                    <DropdownMenuContent className="w-(--radix-dropdown-menu-trigger-width) min-w-56 rounded-lg bg-[#4c1286] border-1 border-[#4c1286]" side={isMobile ? "bottom" : "right"} align="end" sideOffset={15}>
                        <DropdownMenuLabel className="p-0 font-normal">
                            <div className="flex flex-row items-center gap-2 px-1 py-1.5 text-left text-sm">
                                <Avatar className="h-8 w-8 rounded-lg text-white">
                                    <User/>
                                </Avatar>
                                <div className="grid flex-1 text-left text-sm leading-tight text-white">
                                    <span className="truncate font-medium">{user?.name || "Usuário"}</span>
                                    <span className="truncate text-xs">{user?.email || "user@gmail.com"}</span>
                                </div>
                            </div>
                        </DropdownMenuLabel>
                        <DropdownMenuSeparator />
                        <DropdownMenuSeparator />
                        <DropdownMenuItem onClick={handleLogout} className="bg-[#4c1286] text-white cursor-pointer hover:bg-white hover:text-[#4c1286]">
                            <LogOut/>Log out
                        </DropdownMenuItem>
                    </DropdownMenuContent>
                </DropdownMenu>
            </SidebarMenuItem>
        </SidebarMenu>
    )
}

export default NavUser;
