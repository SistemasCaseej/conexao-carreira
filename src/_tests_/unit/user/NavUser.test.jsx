/**
 * @jest-environment jsdom
 */


import {fireEvent, render, screen} from "@testing-library/react";
import NavUser from "@/components/NavUser";
import '@testing-library/jest-dom'


jest.mock('next/navigation', () => ({
    useRouter: () => ({
        push: jest.fn(),
        replace: jest.fn(),
        prefetch: jest.fn(),
        back: jest.fn(),
    }),
}));

jest.mock('/src/components/ui/sidebar', () => ({
    __esModule: true,
    SidebarMenu: (props) => <div data-testid="SidebarMenu" {...props} />,
    SidebarMenuItem: (props) => <div data-testid="SidebarMenuItem" {...props} />,
    SidebarMenuButton: (props) => <button data-testid="SidebarMenuButton" {...props} />,
    useSidebar: () => ({ isMobile: false }),
}));

describe('NavUser Component', ()=> {
    const pushMock = jest.fn();

    const user2 = {
        avatar: 'https://example.com/avatar.png',
        name: 'Test Avatar',
    };

    const user = {
        name: 'José Vitor',
        email: 'jose@example.com',
    };

    it('Renders user name and email correctly', () => {
        render(<NavUser user2={user2} user={user} />);

        expect(screen.getByText('José Vitor')).toBeInTheDocument();
        expect(screen.getByText('jose@example.com')).toBeInTheDocument();
    });

    it('Renders dropdown trigger and responds to click', async () => {
        render(<NavUser user={user} user2={user2} />);

        const triggerButton = screen.getByTestId('SidebarMenuButton');
        fireEvent.click(triggerButton);
    });

    it('Matches Snapshot', () => {
        const { container } = render(<NavUser user={user} user2={user2} />);
        expect(container).toMatchSnapshot();
    });
})