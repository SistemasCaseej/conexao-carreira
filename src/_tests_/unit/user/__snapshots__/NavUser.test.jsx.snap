// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NavUser Component Matches Snapshot 1`] = `
<div>
  <div
    data-testid="SidebarMenu"
  >
    <div
      data-testid="SidebarMenuItem"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        class="data-[state=open]:bg-sidebar-accent data-[state=open]:text-sidebar-accent-foreground"
        data-slot="dropdown-menu-trigger"
        data-state="closed"
        data-testid="SidebarMenuButton"
        id="radix-:r4:"
        type="button"
      >
        <span
          class="relative flex size-8 shrink-0 overflow-hidden h-8 w-8 rounded-lg"
          data-slot="avatar"
        >
          <span
            class="bg-muted flex size-full items-center justify-center rounded-lg"
            data-slot="avatar-fallback"
          >
            CN
          </span>
        </span>
        <div
          class="grid flex-1 text-left text-sm leading-tight"
        >
          <span
            class="truncate font-medium"
          >
            José Vitor
          </span>
          <span
            class="truncate text-xs"
          >
            jose@example.com
          </span>
        </div>
        <svg
          class="lucide lucide-chevrons-up-down ml-auto size-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m7 15 5 5 5-5"
          />
          <path
            d="m7 9 5-5 5 5"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`NavUser Component Matches snapshot 1`] = `
<div>
  <div
    data-testid="SidebarMenu"
  >
    <div
      data-testid="SidebarMenuItem"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        class="data-[state=open]:bg-sidebar-accent data-[state=open]:text-sidebar-accent-foreground"
        data-slot="dropdown-menu-trigger"
        data-state="closed"
        data-testid="SidebarMenuButton"
        id="radix-:r0:"
        type="button"
      >
        <span
          class="relative flex size-8 shrink-0 overflow-hidden h-8 w-8 rounded-lg"
          data-slot="avatar"
        >
          <span
            class="bg-muted flex size-full items-center justify-center rounded-lg"
            data-slot="avatar-fallback"
          >
            CN
          </span>
        </span>
        <div
          class="grid flex-1 text-left text-sm leading-tight"
        >
          <span
            class="truncate font-medium"
          >
            José Vitor
          </span>
          <span
            class="truncate text-xs"
          >
            jose@example.com
          </span>
        </div>
        <svg
          class="lucide lucide-chevrons-up-down ml-auto size-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m7 15 5 5 5-5"
          />
          <path
            d="m7 9 5-5 5 5"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
